### PR TITLE
feat: as support as-configuration data source

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:
as support as configuration data source
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run TestAccDataSourceASConfiguration_basic'
...
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccDataSourceASConfiguration_basic -timeout 360m -parallel 4 
=== RUN   TestAccDataSourceASConfiguration_basic 
=== PAUSE TestAccDataSourceASConfiguration_basic 
=== CONT  TestAccDataSourceASConfiguration_basic
--- PASS: TestAccDataSourceASConfiguration_basic (24.81s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        24.873s
```

### CODE REVIEW

File: docs/data-sources/as_configurations.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  The [object](#instance_config_object) structure is documented below.
+
+* `status` - The AS configuration status, the value can be **Bound** or **Unbound**.
+
+<a name="instance_config_object"></a>
+The `instance_config` block supports:
+
+* `instance_id` - The ECS instance ID when using its specification as the template to create AS configurations.
+
+* `flavor` - The ECS flavor name.
```
Comment (houpeng80) [inline docs/data-sources/as_configurations.md:47]: it is better to modify it to flavor_name

File: docs/data-sources/as_configurations.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+* `image` - The ECS image ID.
```
Comment (houpeng80) [inline docs/data-sources/as_configurations.md:49]: it is better to modify it to image_id

File: huaweicloud/services/as/data_source_huaweicloud_as_configurations.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
```
Comment (houpeng80) [inline huaweicloud/services/as/data_source_huaweicloud_as_configurations.go:136]: here you can extract the contents of Elem into a subfunction, which makes it look simpler.
Such as the resource of huaweicloud_as_bandwidth_policy

File: huaweicloud/services/as/data_source_huaweicloud_as_configurations.go [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+		return diag.Errorf("error getting AS Configuration list: %s", err)
+	}
+
+	configurationList, err := page.(configurations.ConfigurationPage).Extract()
+	if err != nil {
+		return diag.Errorf("error extract to AS Configuration list: %s", err)
+	}
+
+	if len(configurationList) == 0 {
+		return fmtp.DiagErrorf("Your query returned no results. " +
```
Comment (houpeng80) [inline huaweicloud/services/as/data_source_huaweicloud_as_configurations.go]: replaced the package fmtp with fmt

File: huaweicloud/services/as/data_source_huaweicloud_as_configurations.go [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+		elements = append(elements, configurationMap)
+	}
+
+	d.SetId(hashcode.Strings(ids))
+	mErr := multierror.Append(nil,
+		d.Set("configurations", elements),
+		d.Set("region", conf.GetRegion(d)),
+	)
+	if mErr.ErrorOrNil() != nil {
+		return fmtp.DiagErrorf("error setting AS Configuration fields: %s", mErr)
```
Comment (houpeng80) [inline huaweicloud/services/as/data_source_huaweicloud_as_configurations.go]: ditto

File: huaweicloud/services/as/data_source_huaweicloud_as_configurations.go [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+			"status":                     normalizeConfigurationStatus(configuration.ScalingGroupID),
+		}
+		ids = append(ids, configuration.ID)
+		elements = append(elements, configurationMap)
+	}
+
+	d.SetId(hashcode.Strings(ids))
+	mErr := multierror.Append(nil,
+		d.Set("configurations", elements),
+		d.Set("region", conf.GetRegion(d)),
```
Comment (houpeng80) [inline huaweicloud/services/as/data_source_huaweicloud_as_configurations.go]: the param of region has been otained, it is better to use it directly here

File: huaweicloud/services/as/data_source_huaweicloud_as_configurations.go (hunk 3 above)
Comment (deer-hang) [reply huaweicloud/services/as/data_source_huaweicloud_as_configurations.go:136]: OK

File: huaweicloud/services/as/data_source_huaweicloud_as_configurations.go (hunk 4 above)
Comment (deer-hang) [reply huaweicloud/services/as/data_source_huaweicloud_as_configurations.go]: OK

File: huaweicloud/services/as/data_source_huaweicloud_as_configurations.go (hunk 5 above)
Comment (deer-hang) [reply huaweicloud/services/as/data_source_huaweicloud_as_configurations.go]: OK

File: huaweicloud/services/as/data_source_huaweicloud_as_configurations.go (hunk 6 above)
Comment (deer-hang) [reply huaweicloud/services/as/data_source_huaweicloud_as_configurations.go]: OK

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run TestAccDataSourceASConfiguration_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccDataSourceASConfiguration_basic -timeout 360m -parallel 4 
=== RUN   TestAccDataSourceASConfiguration_basic 
=== PAUSE TestAccDataSourceASConfiguration_basic
=== CONT  TestAccDataSourceASConfiguration_basic
--- PASS: TestAccDataSourceASConfiguration_basic (23.23s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        23.292s 
```

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run TestAccDataSourceASConfiguration_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccDataSourceASConfiguration_basic -timeout 360m -parallel 4 
=== RUN   TestAccDataSourceASConfiguration_basic 
=== PAUSE TestAccDataSourceASConfiguration_basic
=== CONT  TestAccDataSourceASConfiguration_basic
--- PASS: TestAccDataSourceASConfiguration_basic (25.99s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        26.031s
```

File: huaweicloud/services/as/data_source_huaweicloud_as_configurations.go [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+				Optional:    true,
+				Description: "The AS image id used to query configuration list.",
+			},
+			"configurations": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "The information about AS instance configurations.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"scaling_configuration_name": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/as/data_source_huaweicloud_as_configurations.go:42]: how about named this parameter as **name**?

File: huaweicloud/services/as/data_source_huaweicloud_as_configurations.go [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+									"security_group_ids": {
+										Type:        schema.TypeList,
+										Computed:    true,
+										Elem:        &schema.Schema{Type: schema.TypeString},
+										Description: "An array of one or more security group IDs.",
+									},
+									"charging_mode": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "Whether the bandwidth is billed by traffic or by bandwidth size.",
```
Comment (ShiChangkuo) [inline huaweicloud/services/as/data_source_huaweicloud_as_configurations.go]: it seems that the description is not correct, please check it

File: huaweicloud/services/as/data_source_huaweicloud_as_configurations.go (hunk 7 above)
Comment (ShiChangkuo) [inline huaweicloud/services/as/data_source_huaweicloud_as_configurations.go:42]: how about named it to **name**?

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run TestAccDataSourceASConfiguration_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccDataSourceASConfiguration_basic -timeout 360m -parallel 4 
=== RUN   TestAccDataSourceASConfiguration_basic 
=== PAUSE TestAccDataSourceASConfiguration_basic 
=== CONT  TestAccDataSourceASConfiguration_basic 
--- PASS: TestAccDataSourceASConfiguration_basic (16.73s) 
PASS 
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        16.796s 
```

**Pull request merged**